bundler/src/main/java/com/f2prateek/bundler/FragmentBundleBuilder.java

```diff
@@ -0,0 +1,419 @@
+package com.f2prateek.bundler;
+
+import android.app.Fragment;
+import android.os.Bundle;
+import android.os.Parcelable;
+import android.util.SparseArray;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+
+/**
+ * A bundle builder for fragment arguments.
+ *
+ * @param <F> The fragment type you're bundling to.
+ */
+@SuppressWarnings("unchecked")
+class FragmentBundleBuilder<F extends Fragment> extends Bundler {
```

Comment:
Hmm...that is a good point, but users wouldn't really ever use this class as a `Bundler`, and so the inheritance doesn't serve a purpose other than reducing code.

```java
 public FragmentBundleBuilder<F> put(String key, boolean value) {
    return (FragmentBundleBuilder<F>) super.put(key, value);
}
```

```java
 public FragmentBundleBuilder<F> put(String key, boolean value) {
   bundle.put(key, value);
   return this;
}
```

I don't think it's that much extra code we're saving, and we can also avoid the ugliness of the casting if we wrap a Bundler (or a `Bundle`).